tests/flow/test_graph_deletion.py

```diff
@@ -244,3 +244,40 @@ def test13_delete_path_elements(self):
         actual_result = redis_graph.query(query)
         self.env.assertEquals(actual_result.nodes_deleted, 1)
         self.env.assertEquals(actual_result.relationships_deleted, 1)
+
+    # Verify that variable-length traversals in each direction produce the correct results after deletion.
+    def test14_post_deletion_traversal_directions(self):
+        self.env.flush()
+        redis_con = self.env.getConnection()
+        redis_graph = Graph("G", redis_con)
+
+        nodes = {}
+        # Create entities.
+        labels = ["Dest", "Src", "Src2"]
+        for idx, l in enumerate(labels):
+            node = Node(label=l, properties={"val": idx})
+            redis_graph.add_node(node)
+            nodes[l] = node
+
+        edge = Edge(nodes["Src"], "R", nodes["Dest"])
+        redis_graph.add_edge(edge)
+        edge = Edge(nodes["Src2"], "R", nodes["Dest"])
+        redis_graph.add_edge(edge)
+        redis_graph.commit()
+
+        # Delete a node.
+        query = """MATCH (n:Src2) DELETE n"""
+        actual_result = redis_graph.query(query)
+        self.env.assertEquals(actual_result.nodes_deleted, 1)
+        self.env.assertEquals(actual_result.relationships_deleted, 1)
+
+        query = """MATCH (n1:Src)-[*]->(n2:Dest {val: 0}) RETURN COUNT(*)"""
+        actual_result = redis_graph.query(query)
+        expected_result = [[1]]
+        self.env.assertEquals(actual_result.result_set, expected_result)
+
+        # Perform the same traversal in the opposite direction.
+        query = """MATCH (dest:Dest {val: 0})<-[*]-(n2:Src) RETURN COUNT(*)"""
```

Comment:
Query pattern wise the direction seemed reversed but the tow patterns are basically the same: src to dest,
if possible use the same aliases as in the previous query. `n1` and `n2`.
we can force transpose if we remove the filter from destination node in one of the queries.